aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,450 @@
+      LOG.error("Cannot drop table {} because table not found or not accessible", identifier, e);
+      return false;
+    } catch (Exception e) {
+      LOG.error("Cannot complete drop table operation for {} due to unexpected exception", identifier, e);
+      throw e;
+    }
+  }
+
+  /**
+   * Rename table in Glue is a drop table and create table.
+   * @param from identifier of the table to rename
+   * @param to new table name
+   */
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    // check new namespace exists
+    if (!namespaceExists(to.namespace())) {
+      throw new NoSuchNamespaceException("Cannot rename %s to %s because namespace %s does not exist",
+          from, to, to.namespace());
+    }
+    // keep metadata
+    Table fromTable = null;
+    String fromTableDbName = IcebergToGlueConverter.getDatabaseName(from);
+    String fromTableName = IcebergToGlueConverter.getTableName(from);
+    String toTableDbName = IcebergToGlueConverter.getDatabaseName(to);
+    String toTableName = IcebergToGlueConverter.getTableName(to);
+    try {
+      GetTableResponse response = glue.getTable(GetTableRequest.builder()
+          .catalogId(awsProperties.glueCatalogId())
+          .databaseName(fromTableDbName)
+          .name(fromTableName)
+          .build());
+      fromTable = response.table();
+    } catch (EntityNotFoundException e) {
+      throw new NoSuchTableException(e, "Cannot rename %s because the table does not exist in Glue", from);
+    }
+
+    // use the same Glue info to create the new table, pointing to the old metadata
+    TableInput.Builder tableInputBuilder = TableInput.builder()
+        .owner(fromTable.owner())
+        .tableType(fromTable.tableType())
+        .parameters(fromTable.parameters())
+        .storageDescriptor(fromTable.storageDescriptor());
+
+    glue.createTable(CreateTableRequest.builder()
+        .catalogId(awsProperties.glueCatalogId())
+        .databaseName(toTableDbName)
+        .tableInput(tableInputBuilder.name(toTableName).build())
+        .build());
+    LOG.info("created rename destination table {}", to);
+
+    try {
+      dropTable(from, false);
+    } catch (Exception e) {
+      // rollback, delete renamed table
+      LOG.error("Fail to drop old table {} after renaming to {}, rollback to use the old table", from, to, e);
+      glue.deleteTable(DeleteTableRequest.builder()
+          .catalogId(awsProperties.glueCatalogId())
+          .databaseName(toTableDbName)
+          .name(toTableName)
+          .build());
+      throw e;
+    }
+
+    LOG.info("Successfully renamed table from {} to {}", from, to);
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    try {
+      glue.createDatabase(CreateDatabaseRequest.builder()
+          .catalogId(awsProperties.glueCatalogId())
+          .databaseInput(IcebergToGlueConverter.toDatabaseInput(namespace, metadata))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;metadata&#39; (index: 1 | type: Map). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;namespace&#39;. The same parameter type is validated here for example: arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java:361. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).